src/OrbitBase/include/OrbitBase/Append.h

```diff
@@ -2,16 +2,32 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <initializer_list>
+#include <type_traits>
 #include <vector>
 
 #ifndef ORBIT_BASE_APPEND_H_
 #define ORBIT_BASE_APPEND_H_
 
 namespace orbit_base {
 
-template <class T>
-void Append(std::vector<T>& dest, const std::vector<T>& source) {
-  dest.insert(std::end(dest), std::begin(source), std::end(source));
+// Append adds the elements given by the range in the second argument to the back of the vector
+// given in the first argument. The range can be anything that defines `Range::value_type` and has
+// cbegin and cend iterators.
+template <class T, class Range,
+          class = typename std::enable_if_t<std::is_convertible_v<typename Range::value_type, T>>>
+void Append(std::vector<T>& dest, const Range& source) {
+  using std::cbegin;
+  using std::cend;
```

Comment:
I guess these `using`s are the fallback in the lookup of `cbegin` and `cend` in `insert`? Or am I off track?

Reply:
Yes. Typical meta programming shenanigans. Assuming Range is of type `my_ns::Range`, then the compiler will first look for `cbegin` in `my_ns` (ADL) and if not found fall back to `std::cbegin`. `std::cbegin` usually tries to call the member function `cbegin`, except in some cases that don't have member functions like C-style arrays.

Reply:
Thanks! I'm happy I got it somewhat right :D